Source/Core/DolphinQt2/Config/ControllersWindow.cpp

```diff
@@ -508,7 +515,11 @@ void ControllersWindow::SaveSettings()
     const std::optional<SerialInterface::SIDevices> si_device = FromGCMenuIndex(index);
     if (si_device)
       SConfig::GetInstance().m_SIDevice[i] = *si_device;
+
     m_gc_buttons[i]->setEnabled(index != 0 && index != 6);
+
+    if (Core::IsRunning())
```

Comment:
You should probably also make sure that `si_device` is valid before using it, no?